extras/kdf-keys/pbkdf2.c

```diff
@@ -43,6 +43,9 @@
 
 #include <gcrypt.h>
 
+/* Max password size */
+#define BUFFER_SIZE 1024
```

Comment:
Why hardcoding the length of the array? (Disregarding the argument, that nobody will have a PW with 1023 chars :D).
Just curious, as this seems a case, where dynamic allocation would be used

Reply:
Sure that no one will have a password with 1024 characters :) That's the reason i set it up to that size. Just that I thought that it's simpler to have a predefined size and to check for known limits. 
We could though use dynamic allocation if we get the size of the password as a parameter, what do you think about that?

Reply:
Yeah, I suppose it could be reduced, although who knows :D Maybe there are people with really long passwords.
Regarding dynamic allocation I can think of three setups, but dunno about costs or good practice
  * Set BUFFER to 20/30/40 or sth like that and allocate array. Run through the reading process, and if BUFFER is exceeded, realloc with 2*BUFFER and so on.
  * No BUFFER at all. Reading password and char for char a realloc (although I suppose this will be the slowest and the logic behind.. well)
 * Process password, deduce length, set array and process again (similar stupid like the second, because two times the work)